src/backend/distributed/utils/node_metadata.c

```diff
@@ -1091,11 +1093,17 @@ AddNodeMetadata(char *nodeName, int32 nodePort,
 	}
 
 	/* user lets Citus to decide on the group that the newly added node should be in */
-	if (nodeMetadata->groupId == 0)
+	if (nodeMetadata->groupId == INVALID_GROUP_ID)
 	{
 		nodeMetadata->groupId = GetNextGroupId();
 	}
 
+	/* if this is a coordinator, we shouldn't place shards on it */
+	if (nodeMetadata->groupId == 0)
+	{
+		nodeMetadata->shouldHaveShards = false;
```

Comment:
the field name `shouldHaveShards ` sounds incomplete now. We'll have reference table shards on the coordinator. But I think it is not easy to come up with a nicer name ?

Reply:
Maybe `shouldHaveDistTablePlacements`, but changing this will probably need a change in `pg_dist_node`, so I think I'd keep for a separate PR.

Reply:
Makes sense, let's keep it for now. There is already a comment next to the field definition.